### PR TITLE
mcobbett maven uses common code

### PR DESCRIPTION
**Pull request by techcobweb**
- maven plugin to use common code.
- avoid junit vulnerability

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-nfcrr for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-fsfjq for details. If you are unable to do so, please contact a member of the Galasa team.

File: galasa-maven-plugin/pom.xml [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 	<groupId>dev.galasa</groupId>
 	<artifactId>galasa-maven-plugin</artifactId>
-	<version>0.32.0</version>
 	<packaging>maven-plugin</packaging>
+	<version>0.32.0</version>
```
Comment (jadecarino) [inline galasa-maven-plugin/pom.xml:10]: 0.33.0

File: galasa-maven-plugin/pom.xml [hunk 2]
```diff
@@ -78,107 +78,231 @@
...
+				<artifactId>dev.galasa.plugin.common.impl</artifactId>
+				<version>0.32.0</version>
+			</dependency>
+
+			<dependency>
+				<groupId>dev.galasa</groupId>
+				<artifactId>dev.galasa.plugin.common.test</artifactId>
+				<version>0.32.0</version>
+				<scope>test</scope>
+			</dependency>
```
Comment (jadecarino) [inline galasa-maven-plugin/pom.xml:100]: Versions need adjusting

File: galasa-maven-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+Environment variable over-rides:
+
+- `LOGS_DIR` - Optional. Where logs are placed. Defaults to creating a temporary directory.
+- `SOURCE_MAVEN` - Optional. Where a maven repository is from which the build will draw artifacts.
+- `DEBUG` - Optional. Defaults to 0 (off)
+- `GPG_PASSPHRASE` - Used to sign and verify artifacts during the build
+
+## How to use the plugin
+
+### Building a test catalog for a Java bundle
```
Comment (jadecarino) [inline galasa-maven-plugin/README.md:17]: These headings haven't rendered for some reason

File: galasa-maven-plugin/README.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+Goal: `deploytestcat`
+
+Phase: `deploy`
+
+Input Parameters/Properties:
+- `galasa.test.stream` required. A string.
+- `galasa.token` optional. An access token for the galasa ecosystem, if that ecosytem is using authentication.
+- `galasa.bootstrap` required. A URL to the ecosystem.
+- `galasa.skip.bundletestcatalog` optional. A boolean. If set to true, the test catalog is not deployed to the Galasa server.
+- `galasa.skip.deploytestcatalog` optional. A boolean. If set to true, the test catalog is not deployed to the Galasa server.
```
Comment (jadecarino) [inline galasa-maven-plugin/README.md:66]: Unsure what the difference between these two are, I was meant to ask you last week. Can we discuss?

**Review by jadecarino — CHANGES_REQUESTED** [review]
Happy to approve once the versions have been merged and comment about bundle vs deploy addressed

File: galasa-maven-plugin/README.md (hunk 4 above)
Comment (jadecarino) [reply galasa-maven-plugin/README.md:66]: Is the description meant to be different on these two lines?

File: galasa-maven-plugin/pom.xml (hunk 2 above)
Comment (techcobweb) [reply galasa-maven-plugin/pom.xml:100]: done.

File: galasa-maven-plugin/README.md (hunk 4 above)
Comment (techcobweb) [reply galasa-maven-plugin/README.md:66]: Explained more in the readme.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-4cqk9 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-ddrnx for details. If you are unable to do so, please contact a member of the Galasa team.

File: galasa-maven-plugin/pom.xml (hunk 1 above)
Comment (techcobweb) [reply galasa-maven-plugin/pom.xml]: Fixed.

File: galasa-maven-plugin/pom.xml (hunk 2 above)
Comment (techcobweb) [reply galasa-maven-plugin/pom.xml]: fixed.

File: galasa-maven-plugin/README.md (hunk 3 above)
Comment (techcobweb) [reply galasa-maven-plugin/README.md:17]: Fixed

File: galasa-maven-plugin/pom.xml (hunk 2 above)
Comment (techcobweb) [reply galasa-maven-plugin/pom.xml]: Can't mark this conversation as resolved.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-jw8bh for details. If you are unable to do so, please contact a member of the Galasa team.

**Review by jadecarino — APPROVED** [review]

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-nsrmk for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-maven-pr-115-48bzh for details. If you are unable to do so, please contact a member of the Galasa team.

**Pull request merged**